test/workspace.js

```diff
@@ -0,0 +1,142 @@
+    code.expect(space.current.deref()).to.equal('function helloWorld(hello){\n  hello = "world";\n}');
+    done();
+  });
+
+  lab.test('#current allows updating current data through .update', function(done){
+    var newText = 'function helloWorld(hello){\n  foo = "bar";\n}';
+
+    code.expect(space.current.update).to.be.a.function();
+
+    space.current.update(function(){
+      return newText;
+    });
+
+    code.expect(space.current.deref()).to.equal(newText);
+    done();
+  });
+
+  lab.test('#saveFile will mkdirp and save a file', function(done){
+    var newText = 'function helloWorld(hello){\n  foo = "bar";\n}';
+
+    space.saveFile(tmpFilepath, newText, function(err){
+      var saved = fs.readFileSync(tmpFilepath, 'utf8');
+
+      code.expect(err).to.not.exist();
+      code.expect(saved).to.equal(newText);
+      done(err);
+    });
+  });
+
+  lab.test('#saveFile also handles cursors', function(done){
+    var newText = 'function helloWorld(hello){\n  my = "cursor";\n}';
+
+    space.current.update(function(){
+      return newText;
+    });
+
+    space.saveFile(tmpFilepath, space.current, function(err){
+      var saved = fs.readFileSync(tmpFilepath, 'utf8');
+
+      code.expect(err).to.not.exist();
+      code.expect(saved).to.equal(newText);
+      done(err);
+    });
+  });
+
+  lab.test('#saveFile adds the file to the directory listing', function(done){
+    var newText = 'function helloWorld(hello){\n  foo = "bar";\n}';
+
+    space.saveFile(tmpFilepath, newText, function(err){
+      code.expect(err).to.not.exist();
+      code.expect(space.directory.contains(tmpFilepath)).to.equal(true);
+      done(err);
+    });
+  });
+
+  lab.test('#saveFile does not add the file to the directory listing twice upon 2nd save', function(done){
+    var newText = 'function helloWorld(hello){\n  foo = "bar";\n}';
+
+    space.saveFile(tmpFilepath, newText, function(err){
+      code.expect(err).to.not.exist();
+      space.saveFile(tmpFilepath, newText, function(err2){
+        code.expect(err2).to.not.exist();
+        code.expect(space.directory.indexOf(tmpFilepath)).to.equal(space.directory.lastIndexOf(tmpFilepath));
+        done(err2);
+      });
+    });
+  });
+
+  lab.test('#saveFile can take workspace.current', function(done){
+    space.saveFile(tmpFilepath, space.current, function(err){
+      var saved = fs.readFileSync(tmpFilepath, 'utf8');
+
+      code.expect(err).to.not.exist();
+      code.expect(space.current.deref()).to.equal(saved);
+      done(err);
+    });
+  });
+
+  lab.test('#loadFile will load a file into current', function(done){
+    space.loadFile(tmpFilepath, function(err, contents){
+      code.expect(err).to.not.exist();
+      code.expect(contents).to.equal(space.current);
+      done();
+    });
+  });
+
+  lab.test('#directory should be a list of files in a directory', function(done){
+    code.expect(space.directory.deref).to.be.a.function();
```

Comment:
or maybe we should read the files in upon startup.  problem is that the operation would be async